services/scripts.py

```diff
@@ -1,15 +1,49 @@
 # Music recommendation application
 import csv
-from .cfg import ARTIST_INDEX, LIKES_INDEX
+
+from cfg import TableEnum
+
+
+class Song:
+    def __init__(self, row: list[str]) -> None:
+        self.id = int(row[TableEnum.ID.value])
+        self.name = row[TableEnum.Track.value]
+        self.album = row[TableEnum.Album.value]
+        self.artist = row[TableEnum.Artist.value]
+
+
+class DataBase:
+    def __init__(self, filename):
+        self.db = {}
+
+        with open(filename, encoding='utf-8-sig') as file:
+            _ = file.readline()
+            reader = csv.reader(file)
+
+            for row in reader:
+                self.add_song(Song(row))
+
+    def add_song(self, song: Song) -> None:
+        self.db[song.id] = song
+
+    def get_song(self, song_id: int) -> Song:
```

Comment:
The problem of this method is that if you want to work with some other classes in a future, you will need to add methods with the same functionality but for different classes, it is better to pass in a class obj from which you want to get a result. So you should consider adding table obj to database, than passing in an object of this table when running a get method. Also, consider taking as an example this API:[https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#built-in-from-linting-will-warn-for-any-potential-cartesian-products-in-a-select-statement] Here you can see that session.query takes in an object of a table an only than it executes some filter method, in our case this will be get method.

Reply:
first of all, [here](https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#built-in-from-linting-will-warn-for-any-potential-cartesian-products-in-a-select-statement) is a clickable link, secondly, it seems to me that our project is a little bit smaller than sqlalchemy… yeah, `query` is a really nice thing, but, as far as i know, our `DataBase` class is focused on storing a songs with static format (what is required by the condition in the [README.md](https://github.com/D1ffic00lt/DSBA-Python-2024/blob/main/README.md)). from this we can suggest changing the class name to, maybe, `SongsDataBase`, but in my opinion it is also useless

Reply:
Frankly speaking, this class is a _data_ manager or a song collection. It's difficult to call it a **"DataBase"**.

Reply:
But what should I change?